runbot/models/build_error.py

```diff
@@ -497,19 +497,28 @@ def action_view_analogous_qualified_contents(self):
         }
 
     def action_assign(self):
-        teams = None
-        repos = None
-        for record in self:
-            if not record.responsible and not record.team_id:
-                for error_content in record.error_content_ids:
-                    if error_content.file_path:
-                        if teams is None:
-                            teams = self.env['runbot.team'].search(['|', ('path_glob', '!=', False), ('module_ownership_ids', '!=', False)])
-                            repos = self.env['runbot.repo'].search([])
-                        team = teams._get_team(error_content.file_path, repos)
-                        if team:
-                            record.team_id = team
-                            break
+        teams = lazy(self.env['runbot.team'].search, ['|', ('path_glob', '!=', False), ('module_ownership_ids', '!=', False)])
+        repos = lazy(self.env['runbot.repo'].search, [])
```

Comment:
I've started experimenting in my branch with a pattern that could be used in this case too. 

in team.py and repo.py
```python
    def _get_ownership_teams(self):
        if '_get_ownership_teams' not in self.env.cr.cache:
             self.env.cr.cache['_get_error_auto_merge'] = self.env['runbot.team'].search(['|', ('path_glob', '!=', False), ('module_ownership_ids', '!=', False)])
        return  self.env.cr.cache['_get_error_auto_merge']
        
    def get_all_repo(self):
        if 'get_all_repo' not in self.env.cr.cache:
             self.env.cr.cache['get_all_repo'] = self.env['runbot.repo'].search([])
        return  self.env.get_all_repo.cache['_get_error_auto_merge']
```

It is a transaction cache but more flexible than the record one. Even if not officially supported using a decorator or so, this is the purpose of the cr.cache (kind of)

Then 

```suggestion
        teams = lazy(self.env['runbot.team']._get_ownership_teams)
        repos = lazy(self.env['runbot.repo'].get_all)
```

This will allow to make the search once for all error when scanning multiple of them during the nightly.

Reply:
Note: this could be done in another pr ;)